packages/react-app/src/components/NetworkSelector.jsx

```diff
@@ -1,86 +1,74 @@
+import {
+  Button,
+  Flex,
+  Popover,
+  PopoverBody,
+  PopoverContent,
+  PopoverTrigger,
+  Text,
+} from '@chakra-ui/core';
 import React, { useContext, useEffect, useState } from 'react';
-import Select, { components } from 'react-select';
 
 import { BridgeContext } from '../contexts/BridgeContext';
 import { Web3Context } from '../contexts/Web3Context';
 import { DownArrowIcon } from '../icons/DownArrowIcon';
 import { networkOptions } from '../lib/constants';
 
-const { Option, DropdownIndicator } = components;
+const SelectOption = props => {
+  const { onChange, network } = props;
+  const { icon, label } = network;
 
-const CustomSelectOption = props => {
-  const {
-    data: { icon, label },
-  } = props;
   return (
-    <Option {...props}>
-      <Flex
-        align="center"
-        cursor="pointer"
-        color="grey"
-        transition="0.25s"
-        _hover={{ color: 'blue.500' }}
+    <Flex color="grey" transition="0.25s">
+      <Button
+        background="transparent"
+        width="100%"
+        justifyContent="flex-start"
+        _hover={{ color: 'blue.500', background: 'gray.100', zIndex: 4 }}
+        onClick={() => onChange(network)}
       >
         {icon}
         <Text color="black" ml={2}>
           {label}
         </Text>
-      </Flex>
-    </Option>
+      </Button>
+    </Flex>
   );
 };
 
-const CustomDropdownIndicator = props => {
+const DropdownIndicator = props => {
   return (
-    <DropdownIndicator {...props}>
-      <Flex align="center" justify="center">
-        <DownArrowIcon fontSize={8} color="black" />;
-      </Flex>
-    </DropdownIndicator>
+    <Flex align="center" justify="center" paddingLeft="15px">
+      <DownArrowIcon fontSize={8} color="black" />
+    </Flex>
   );
 };
 
-const CustomSelectValue = ({ data: { icon, label } }) => (
+const SelectValue = ({ icon, label }) => (
   <Flex
-    align="center"
     cursor="pointer"
     color="grey"
     transition="0.25s"
+    padding="2px 8px"
     _hover={{ color: 'blue.500' }}
   >
     {icon}
-    <Text color="black" ml={2}>
+    <Text color="black" ml={2} fontWeight="normal">
       {label}
     </Text>
+    <DropdownIndicator />
   </Flex>
 );
 
-const customStyles = {
-  control: provided => ({
-    ...provided,
-    cursor: 'pointer',
-    border: 'none',
-    background: 'transparent',
-    fontWeight: 'bold',
-    paddingLeft: '0.5rem',
-  }),
-  menu: provided => ({
-    ...provided,
-    boxShadow: '0 0.5rem 1rem #CADAEF',
-    backgroundColor: 'white',
-  }),
-  indicatorSeparator: provided => ({
-    ...provided,
-    backgroundColor: 'transparent',
-  }),
-};
-
 export const NetworkSelector = props => {
   const [localNetwork, setLocalNetwork] = useState(0);
   const { setNetwork } = useContext(Web3Context);
   const { setDefaultToken } = useContext(BridgeContext);
 
+  const [isOpen, setIsOpen] = React.useState(false);
+  const open = () => setIsOpen(true);
+  const close = () => setIsOpen(false);
```

Comment:
I took over the native open/close of the popover so the popover exits when an option is chosen.